### ISSUE TITLE
Clean up old log4net log files

### ISSUE THREAD

**Issue by JuliusSweetland** [OptiKey/OptiKey#171]
At the moment OptiKey keeps an infinite number of log files. This is controlled by the log4net.config file in the root of the project. Logs are output to (on my computer): C:\Users\Julius\AppData\Roaming\JuliusSweetland\OptiKey\Logs

**Comment by SamPlusPlus** [comment]
@JuliusSweetland I could probably take look at this, but can you elaborate on the desired behavior. 

For example I tend to do something like 3 rolling files up to 5MB each. Do you have a preference?

**Comment by JuliusSweetland** [comment]
Hi Samuel. Great, thanks. I think the desired behaviour would be one file
per run (which is what I was trying to achieve with the filename mask) and
a maximum of x files to correspond to the last a runs. For arguments sake
I'd say 10 log files should be stored. I don't think file size is
particularly important, but please advise if you have other ideas.

**Comment by SamPlusPlus** [comment]
Just curious, why a log per a run?  Doesn't that make is harder on the user to send you a log file. For example. "Hi Julius, I keep getting an exception randomly but have no idea why it happens" if you ask the user for the log we won't know in which run it happened, or how many times they have run the application since their last crash"

Would it not be better to just have 3 rolling logs of  say 5MB's each, at any given time you could ask the user for 15mb's of text and log information, that should be plenty, and then you could search for the exception. You could also have the application write to the log that it started up.

I think this use case is supported by Log4Net by default, without any special logic. 

In other words whats the reason per a log file per an instance run?

**Comment by JuliusSweetland** [comment]
If OptiKey crashes the latest log is automatically emailed over. It needs
to be as small as possible really.

**Comment by SamPlusPlus** [comment]
@JuliusSweetland I figured you had a reason. Thanks for explaining, I will look into this.

**Comment by SamPlusPlus** [comment]
Looks like I need to update my machine to support this. You using Visual Studio 2015 with .net 4.6?

**Comment by JuliusSweetland** [comment]
.net 4.6, and either visual studio 2012, 2013, or 2015

**Comment by NikitasKotsolakos** [comment]
Hello. First of all, just wanted you to know that you are running a great project and it is the one that finally inspired me to start contributing to open source.

The reason the logfiles are not limited to the number specified in maxSizeRollBackups  is because the date pattern in the filename makes the maxSizeRollBackups property to not work correctly. I removed the date file name and it works fine, see my pull request. (It is still creating one log per application run)

Please note, that since I am not experienced with NBug and do not know how it sends you the latest logfile on crash, perhaps you will also have to make some changes there if it does so by logfile name (or let me know where that is and some basic info, and I can do it).

Finally, if for some reason you really need to have date in the logfile name, let me know to search for another solution.

**Comment by EddPorter** [comment]
I think NBug should be fine with your change, as it just sends the current contents of the log file. This happens in the `AttachUnhandledExceptionHandlers` method in `App.xaml.cs` if you're interested..

As you mentioned, @NikitasKotsolakos, log4net doesn't support limiting the number of log files kept if you roll based on date/times: http://logging.apache.org/log4net/release/sdk/html/T_log4net_Appender_RollingFileAppender.htm
